### DISCUSSION TITLE
Handle scroll up/down on Time/Date picker to add or subtract hour and minutes

### DISCUSSION THREAD

**Discussion by FluffyDiscord** [solidtime-io/solidtime#244] [Feature requests]
Allow to use mousewheel to adjust hours and minutes. Moving mouse wheel up or down on hours would add or subtract hours and moving on minutes would adjust minutes. 

Same for Date inputs.

![image](https://github.com/user-attachments/assets/cd910b96-e9f9-450b-a896-b0be73de100d)

**Comment by nilaygit-10721** [comment]
can you assign me this issue?

**Comment by FluffyDiscord** [comment]
You can always fork and create a PR

**Comment by Onatcer** [comment]
We do not accept external contributions for now so please don't create a PR for that.

Do you have examples of applications doing that? I have never noticed such behavior, but i might have never tried.

I agree that the date and time selectors are cumbersome to use right now, I have a few ideas how to improve them, but haven't gotten around to it yet. My worries with this is that it might be inconsistent with different mouse/scroll settings, and i also don't like scroll hijacking too much as it can cause weird behaviour when you just want to scroll down, although for the time/datepickers currently i don't think that's too much of a problem.

**Comment by FluffyDiscord** [comment]
> Do you have examples of applications doing that?

Not really, because everyone seems to forget that mouse wheel actually exists and can be used for something useful rather than full page animations, where you actually do hijack the scroll.

You could add a key modifier like CTRL or SHIFT to enable it.

Both Time and Date pickers are cumbersome, but to be honest, I have yet to see a good ones, either they are extremely simple and thus pain in the ass to use because you need to click or tab so much, or so over-engineered that no mortal man can use it without getting angry. It's always a pain to make them perfect.

If you still are against using a mouse wheel, what about option (toggle perhaps) to hide all that picker stuff and show just a single text field, where you would just enter the date and time and system would parse it. I have done this in past, it was a bit harder to achieve and parse all possible variations, but I could just write for example "dd.mm.yyyy hh:mm:ss" and it would parse it, or "dd.mm." and it would parse it and add current year and time (I also added the wheel enhancement, so if you needed to add one day for example, you simply had to hover over the field, press SHIFT + mousewheel up, and be done)

**Comment by FluffyDiscord** [comment]
Any updates? This automatic issue/fr closing isn't great to be honest

**Comment by Onatcer** [comment]
We decided to move feature requests into github discussions, to separate feature requests from bugs and allow for better prioritization. If this feature is important to you, you can upvote it to let us know. 

I'll push an update to the time picker in the next few days which will change the UX and adds a dropdown for 15-minute steps. This is in line with the UX of Google Calendar. If you have any feedback on that let me know. 
<img width="654" alt="image" src="https://github.com/user-attachments/assets/188d3621-1047-4ff6-9350-b0cae9d7b2b5">
